components/sst_importer/src/import_mode.rs

```diff
@@ -1,67 +1,143 @@
 // Copyright 2018 TiKV Project Authors. Licensed under Apache-2.0.
 
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use std::time::Instant;
+
 use engine_traits::{ColumnFamilyOptions, DBOptions, KvEngine};
+use futures_cpupool::CpuPool;
+use futures_util::compat::{Compat, Future01CompatExt};
+use futures_util::future::FutureExt;
 use kvproto::import_sstpb::*;
+use tikv_util::timer::GLOBAL_TIMER_HANDLE;
 
+use super::Config;
 use super::Result;
 
 type RocksDBMetricsFn = fn(cf: &str, name: &str, v: f64);
 
-pub struct ImportModeSwitcher {
+struct ImportModeSwitcherInner<T: KvEngine> {
```

Comment:
```suggestion
struct ImportModeSwitcherInner<E: KvEngine> {
```

TiKV typically uses `E` for the engine.